src/metrics/misc.ts

```diff
@@ -174,16 +164,60 @@ export class MetadataMetric implements Metric {
     const blackPlayer = metadata?.find((data) => data.startsWith('[Black'));
     const whiteUsername = whitePlayer?.split('"')[1];
     const blackUsername = blackPlayer?.split('"')[1];
-    
+
     if (whiteUsername) {
-      this.playerGameStats[whiteUsername] = (this.playerGameStats[whiteUsername] || 0) + 1;
+      this.playerGameStats[whiteUsername] =
+        (this.playerGameStats[whiteUsername] || 0) + 1;
     }
     if (blackUsername) {
-      this.playerGameStats[blackUsername] = (this.playerGameStats[blackUsername] || 0) + 1;
+      this.playerGameStats[blackUsername] =
+        (this.playerGameStats[blackUsername] || 0) + 1;
+    }
+
+    if (metadata[4] === '[Result "1-0"]') {
+      this.whiteWins++;
+    } else if (metadata[4] === '[Result "0-1"]') {
+      this.blackWins++;
+    } else if (metadata[4] === '[Result "1/2-1/2"]') {
+      this.ties++;
+    } else {
+      throw new Error('Invalid result');
     }
 
     // Increment the number of games analyzed
     this.numberGamesAnalyzed++;
-  
   }
-}
\ No newline at end of file
+
+  // Aggregate the results of the metric
+  aggregate() {
+    // Calculate the average player rating after each game
+    // 2 players per game, so need to divide by two given that the totalPlayerRating adds all ratings up from both players
+    this.averagePlayerRating =
+      this.totalPlayerRating / (this.numberGamesAnalyzed * 2);
+
+    // Calculate the average player rating diff
+    this.averageRatingDiff =
+      this.totalPlayerRatingDiff / this.numberGamesAnalyzed;
+
+    // Calculate the player with the most games played
+    let maxGames = 0;
+    let playerMostGames = '';
+    for (const player in this.playerGameStats) {
+      if (this.playerGameStats[player] > maxGames) {
+        maxGames = this.playerGameStats[player];
+        playerMostGames = player;
+      }
+    }
+    this.mostGamesPlayed = maxGames;
+    this.playerMostGames = playerMostGames;
+
+    return {
+      averagePlayerRating: this.averagePlayerRating,
+      averageRatingDiff: this.averageRatingDiff,
+      mostGamesPlayed: maxGames,
+      playerMostGames,
+      // This doesn't account for ties
+      whiteWinRatio: this.whiteWins / (this.whiteWins + this.blackWins),
```

Comment:
`whiteWinRatioWithTies: (this.whiteWins + (this.ties/2)) / (this.whiteWins + this.blackWins + (this.ties / 2))`

to calculate the ratio of white wins to black wins including counting a tie as half a point for both black and white

Reply:
didn't know how we wanted to calc exaclty, introducing any of these is ez

Reply:
merging & we can add l8r